client/components/data/query-terms/index.jsx

```diff
@@ -27,7 +27,7 @@ class QueryTerms extends Component {
 	}
 
 	request( props ) {
-		if ( props.requesting ) {
+		if ( props.requesting || ! props.siteId ) {
```

Comment:
Very true I have used that approach also `siteId && <QueryTerms />` - but even though I authored `<QueryTerms />` I forgot to safeguard its usage 😆  - so I'm going to leave this in place here just in case.